src/lib/frontend/typechecker.ml

```diff
@@ -171,6 +164,10 @@ module Types = struct
             from_labels =
               List.fold_left
                 (fun fl (l,_) -> MString.add l id fl) env.from_labels lbs }
+    | Enum l ->
+      let body = List.map (fun constr -> Uid.of_string constr, []) l in
+      let ty = Ty.t_adt ~enum:true ~body:(Some body) (Uid.of_string id) [] in
+      ty, { env with to_ty = MString.add id ty env.to_ty }
```

Comment:
Not sure why this was moved; the check for `PolymorphicEnum` error should be kept (we do bad things when type variables are unused and this is a case where we can catch it).

Reply:
It's an artifact of the last rebase. I will fix that.